bin/verilator

```diff
@@ -3746,28 +3756,30 @@ Verilator is mostly a two state simulator, not a four state simulator.
 However, it has two features which uncover most initialization bugs
 (including many that a four state simulator will miss.)
 
-Identity comparisons (=== or !==) are converted to standard ==/!== when
-neither side is a constant.  This may make the expression result differ
-from a four state simulator.  An === comparison to X will always be false,
-so that Verilog code which checks for uninitialized logic will not fire.
+Identity comparisons (=== or !==) are converted to standard ==/!= when neither
+side is a constant.  This may make the expression yield a different result
+compared to a four state simulator.  An === comparison to X will always be
+false, so that Verilog code which checks for uninitialized logic will not fire.
 
-Assigning a variable to a X will actually assign the variable to a random
-value (see the --x-assign switch and +verilator+rand+reset simulation
-runtime switch.)  Thus if the value is actually used, the random value
-should cause downstream errors.  Integers also randomize, even though the
-Verilog 2001 specification says they initialize to zero.
+Assigning X to a variable will actually assign a constant value as determined
+by the --x-assign switch.  This allows run-time randomization, thus if the
+value is actually used, the random value should cause downstream errors.
+Integers also get randomized, even though the Verilog 2001 specification says
+they initialize to zero.  Note however that randomization happens at
+initialziation time and hence during a single simulation run, the same constant
+(but random) value will be used every time the assignment is executed.
 
 All variables, depending on --x-initial setting, are typically randomly
 initialized using a function.  By running several random simulation runs
-you can determine that reset is working correctly.  On the first run, the
-function initializes variables to zero.  On the second, have it initialize
+you can determine that reset is working correctly.  On the first run, have the
+function initialize variables to zero.  On the second, have it initialize
 variables to one.  On the third and following runs have it initialize them
 randomly.  If the results match, reset works.  (Note this is what the
-hardware will really do.)  In practice, just setting all variables to one
+hardware will really do).  In practice, just setting all variables to one
```

Comment:
It was correct. https://english.stackexchange.com/questions/6632/where-does-the-period-go-when-using-parentheses